server/game/cards/01_SOR/leaders/BobaFettCollectingTheBounty.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+
+export default class BobaFettCollectingTheBounty extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '4626028465',
+            internalName: 'boba-fett#collecting-the-bounty',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'Ready a resource',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            when: {
+                onCardReturnedToHand: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller && context.source.controller.resources.some((resource) => resource.exhausted),
+                onCardDefeated: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller && context.source.controller.resources.some((resource) => resource.exhausted),
```

Comment:
We have to be intentional about how we set up the exhausted condition here, since it is possible that state will change in between the trigger time and the actual resolution time of the ability. See for example how Jedha Agitator is coded, with a conditional in the effect instead of in the trigger.

The difference is between these two ways of expressing the ability (pretending that the exhaust condition was an actual part of the text):

**When an enemy unit leaves play and you control an exhausted resource**: You may exhaust this leader. If you do, ready a resource.

**When an enemy unit leaves play**: If you control an exhausted resource, you may exhaust this leader. If you do, ready a resource.

In the latter case, the condition isn't checked until the time that the ability resolves. So if there were, for example, a triggered ability in the same window that caused a resource to become exhausted, we could activate Boba1's ability without missing the opportunity.

That said, the downside of doing it that was is that it means if there are multiple triggered abilities in the same window, the player will be prompted to resolve all of them. That includes the Boba1 ability, which may or may not actually have an effect (though the player should be able to realize that).

Lmk in chat if that's not clear. I'm not 100% sure what the best option is, I'm leaning towards thinking the Jedha Agitator way is better but curious for your thoughts